pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/HashUtils.java

```diff
@@ -36,6 +40,31 @@ public static byte[] hashMD5(byte[] bytes) {
     return Hashing.md5().hashBytes(bytes).asBytes();
   }
 
+  /**
+   * Returns a byte array that is a concatenation of the binary representation of each of the passed UUID values. This
+   * is done by getting a String from each value by calling {@link Object#toString()}, which is then used to create a
+   * {@link UUID} object. The 16 bytes of each UUID are appended to a buffer which is then returned in the result.
+   * If any of the values is not a valid UUID, then we return the result of {@link #concatenate}.
+   */
+  public static byte[] hashUUID(Object[] values) {
+    byte[] result = new byte[values.length * 16];
+    ByteBuffer byteBuffer = ByteBuffer.wrap(result).order(ByteOrder.BIG_ENDIAN);
+    for (Object value : values) {
+      if (value == null) {
+        return concatenate(values);
```

Comment:
Silently fallback can cause performance issue and is very hard to debug. I'd suggest not falling back but directly throw exception

Reply:
Stopping ingestion doesn't make sense. We should ideally log every few batch of events.

Reply:
If it throws exception here it basically tells that the data is not UUID. IMO this is similar to user defining a column as LONG but inserting string values. We will throw exception in that scenario and fail the ingestion

Reply:
Users can have outlier data, which may be 0.01% of the total data. I think stopping ingestion is too strict.

Also, we don't follow this convention (of throwing on invalid data) consistently. For instance booleans are auto converted using the logic defined in BooleanUtils.

Reply:
Just realized that we can directly use `PrimaryKey.asBytes()` which compresses the data better